### PR TITLE
Bump to v19.24.6

### PR DESCRIPTION
**Pull request by SfinxCZ**
New pull request to fix the checkout issues - see https://github.com/alesanfra/dlib-wheels/pull/12 for more details.

### CODE REVIEW

**Comment by SfinxCZ** [comment]
Hello @alesanfra, I've modified the pipeline to run on correct commit, can you please approve the workflow so I cen verify that it works correctly?

Thanks

**Comment by SfinxCZ** [comment]
Hello @alesanfra, can you please approve the pipeline again? Thanks.

**Comment by SfinxCZ** [comment]
Hey @alesanfra,
can you please review this pull request? The pipeline passed on correct commit, so LGTM.

Thanks

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,15 +1,17 @@
 name: Build
 
 on:
-  push:
```
Comment (alesanfra) [inline .github/workflows/build.yaml:4]: why did you change the triggers? what's the benefit of the new config?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -20,6 +22,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
```
Comment (alesanfra) [inline .github/workflows/build.yaml:25]: What's the purpose of this line?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (SfinxCZ) [reply .github/workflows/build.yaml:4]: The `pull_request_target` does not run the pipeline on the pull request head commit, it runs on the commit of the base (target) branch - https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target 

This is the reason why the pipeline run on incorrect commit in previous pull request.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (SfinxCZ) [reply .github/workflows/build.yaml:25]: Its related to the previous comment - if you want to run on the pull request head commit, you have to use the `github.event.pull_request.head.sha` otherwise it will run on the main branch from this repository - https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request

**Comment by alesanfra** [comment]
LGTM
